Adding examples, correcting texts, adding links and various contribut… (#40055)
mdn/content: files/en-us/web/http/reference/status/418/index.md

The HTTP **`418 I'm a teapot`** status response code indicates that the server refuses to brew coffee because it is, permanently, a teapot.
A combined coffee/tea pot that is temporarily out of coffee should instead return {{HTTPStatus("503")}}.

> This error is a reference to Hyper Text Coffee Pot Control Protocol, defined as an April Fools' jokes in 1998 and 2014.

Reviewer: ```suggestion
This error is a reference to the Hyper Text Coffee Pot Control Protocol, introduced as April Fools' jokes in 1998 and 2014.
```
